### PR TITLE
Refactor accumulators and hdf5 (addresses #212)

### PR DESCRIPTION
**Pull request by krivenko**
This PR fixes problems listed in https://github.com/ALPSCore/ALPSCore/issues/212#issuecomment-365136271 (to some extent). It drops compilation time of the library from 8 min down to 5 min on my laptop in the `Debug` mode (`make -j4)`.

* Boost.MPL and Boost.TypeTraits have been removed and replaced with metafunctions from `<type_traits>`. The only place where MPL is left is `include/alps/params/dict_types.hpp` (I didn't want to mess with Alex's work).

* A lot of code has been moved from accumulators and hdf5 headers to a bunch of relatively small .cpp files. I am satisfied with the compilation times and the sizes of the resulting object files in most cases. An exception is `accumulators/src/wrapper_set_hdf5.cpp` (19 Mb object file in the `Debug` mode). Unfortunately, I don't see a way to improve the situation without a complete redesign of the hdf5 registration mechanism used in the accumulators.

* I had to sacrifice the support for user-defined observable types (`ALPS_ACCUMULATOR_USER_TYPES` macro and `custom_type` test). This feature was what made accumulators an almost header-only library.

* Huge macros are replaced with templates.

* Lukas Gamper wrote some preliminary code (`accumulators_.hpp`, `accumulators.cpp` and `mean_wrapped` test) trying to refactor the accumulators. That code had never been used, so I excluded it from the build.

* There were broken methods that didn't compile. I had to comment them out. My hypothesis is that nobody ever saw those compilation failures because the methods were never instantiated. 
  - `converged_errors()` of `binning_analysis` accumulator.
  - `covariance()` of `max_num_binning` result.
  - `accurate_covariance()` of `max_num_binning` result.

I could have gone deeper with the refactoring, but that effort would lead to a disproportionate outcome in terms of compilation speed.

If you think this PR breaks too many things, feel free to close it without merging.

### CODE REVIEW

**Comment by kleinhenz** [comment]
This fails to compile for me with clang.

```
../accumulators/src/wrapper_set.cpp:63:63: error: explicit specialization of 'merge' after instantiation
            template<> void wrapper_set<accumulator_wrapper>::merge(wrapper_set const &rhs) {
                                                              ^
../accumulators/include/alps/accumulators/wrapper_set.hpp:92:35: note: explicit instantiation first required here
            extern template class wrapper_set<accumulator_wrapper>;
                                  ^
../accumulators/src/wrapper_set.cpp:72:63: error: explicit specialization of 'reset' after instantiation
            template<> void wrapper_set<accumulator_wrapper>::reset() {
                                                              ^
../accumulators/include/alps/accumulators/wrapper_set.hpp:92:35: note: explicit instantiation first required here
            extern template class wrapper_set<accumulator_wrapper>;
```

**Comment by egull** [comment]
I find Joe's problems, too.

**Comment by krivenko** [comment]
@kleinhenz @egull Please, try again.

**Comment by kleinhenz** [comment]
now I get a linker error

```
FAILED: accumulators/libalps-accumulators.dylib
: && /usr/local/bin/mpicxx -ftemplate-depth=1024 -O3 -DNDEBUG -DBOOST_DISABLE_ASSERTS  -dynamiclib -Wl,-headerpad_max_install_names  -o accumulators/libalps-accumulators.dylib -install_name @rpath/libalps-accumulators.dylib accumulators/CMakeFiles/alps-accumulators.dir/src/namedaccumulators.cpp.o accumulators/CMakeFiles/alps-accumulators.dir/src/accumulator_wrapper.cpp.o accumulators/CMakeFiles/alps-accumulators.dir/src/result_wrapper.cpp.o accumulators/CMakeFiles/alps-accumulators.dir/src/wrapper_set.cpp.o accumulators/CMakeFiles/alps-accumulators.dir/src/wrapper_set_hdf5.cpp.o accumulators/CMakeFiles/alps-accumulators.dir/src/mpi.cpp.o accumulators/CMakeFiles/alps-accumulators.dir/src/feature/count.cpp.o accumulators/CMakeFiles/alps-accumulators.dir/src/feature/mean.cpp.o accumulators/CMakeFiles/alps-accumulators.dir/src/feature/error.cpp.o accumulators/CMakeFiles/alps-accumulators.dir/src/feature/binning_analysis.cpp.o accumulators/CMakeFiles/alps-accumulators.dir/src/feature/max_num_binning.cpp.o  -Wl,-rpath,/Users/Joseph/Documents/physics/research/umich/egull/libraries/ALPSCore/build/hdf5 -Wl,-rpath,/Users/Joseph/Documents/physics/research/umich/egull/libraries/ALPSCore/build/utilities /usr/local/Cellar/hdf5/1.10.1_2/lib/libhdf5.dylib /usr/local/opt/szip/lib/libsz.dylib /usr/lib/libz.dylib /usr/lib/libdl.dylib /usr/lib/libm.dylib hdf5/libalps-hdf5.dylib /usr/local/Cellar/hdf5/1.10.1_2/lib/libhdf5.dylib /usr/local/opt/szip/lib/libsz.dylib /usr/lib/libz.dylib /usr/lib/libdl.dylib /usr/lib/libm.dylib utilities/libalps-utilities.dylib && :
Undefined symbols for architecture x86_64:
  "void alps::alps_mpi::reduce<std::__1::vector<unsigned long long, std::__1::allocator<unsigned long long> >, std::__1::plus<unsigned long long> >(alps::mpi::communicator const&, std::__1::vector<unsigned long long, std::__1::allocator<unsigned long long> > const&, std::__1::vector<unsigned long long, std::__1::allocator<unsigned long long> >&, std::__1::plus<unsigned long long>, int)", referenced from:
      alps::accumulators::impl::Accumulator<float, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<float, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<float, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<float, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<float> > > > >::collective_merge(alps::mpi::communicator const&, int) in binning_analysis.cpp.o
      alps::accumulators::impl::Accumulator<double, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<double, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<double, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<double, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<double> > > > >::collective_merge(alps::mpi::communicator const&, int) in binning_analysis.cpp.o
      alps::accumulators::impl::Accumulator<long double, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<long double, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<long double, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<long double, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<long double> > > > >::collective_merge(alps::mpi::communicator const&, int) in binning_analysis.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<float, std::__1::allocator<float> >, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<std::__1::vector<float, std::__1::allocator<float> >, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<std::__1::vector<float, std::__1::allocator<float> >, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<std::__1::vector<float, std::__1::allocator<float> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<float, std::__1::allocator<float> > > > > > >::collective_merge(alps::mpi::communicator const&, int) in binning_analysis.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<double, std::__1::allocator<double> >, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<std::__1::vector<double, std::__1::allocator<double> >, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<std::__1::vector<double, std::__1::allocator<double> >, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<std::__1::vector<double, std::__1::allocator<double> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<double, std::__1::allocator<double> > > > > > >::collective_merge(alps::mpi::communicator const&, int) in binning_analysis.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<long double, std::__1::allocator<long double> >, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<std::__1::vector<long double, std::__1::allocator<long double> >, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<std::__1::vector<long double, std::__1::allocator<long double> >, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<std::__1::vector<long double, std::__1::allocator<long double> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<long double, std::__1::allocator<long double> > > > > > >::collective_merge(alps::mpi::communicator const&, int) in binning_analysis.cpp.o
  "void alps::alps_mpi::reduce<std::__1::vector<unsigned long long, std::__1::allocator<unsigned long long> >, std::__1::plus<unsigned long long> >(alps::mpi::communicator const&, std::__1::vector<unsigned long long, std::__1::allocator<unsigned long long> > const&, std::__1::plus<unsigned long long>, int)", referenced from:
      alps::accumulators::impl::Accumulator<float, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<float, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<float, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<float, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<float> > > > >::collective_merge(alps::mpi::communicator const&, int) const in binning_analysis.cpp.o
      alps::accumulators::impl::Accumulator<double, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<double, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<double, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<double, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<double> > > > >::collective_merge(alps::mpi::communicator const&, int) const in binning_analysis.cpp.o
      alps::accumulators::impl::Accumulator<long double, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<long double, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<long double, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<long double, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<long double> > > > >::collective_merge(alps::mpi::communicator const&, int) const in binning_analysis.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<float, std::__1::allocator<float> >, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<std::__1::vector<float, std::__1::allocator<float> >, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<std::__1::vector<float, std::__1::allocator<float> >, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<std::__1::vector<float, std::__1::allocator<float> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<float, std::__1::allocator<float> > > > > > >::collective_merge(alps::mpi::communicator const&, int) const in binning_analysis.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<double, std::__1::allocator<double> >, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<std::__1::vector<double, std::__1::allocator<double> >, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<std::__1::vector<double, std::__1::allocator<double> >, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<std::__1::vector<double, std::__1::allocator<double> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<double, std::__1::allocator<double> > > > > > >::collective_merge(alps::mpi::communicator const&, int) const in binning_analysis.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<long double, std::__1::allocator<long double> >, alps::accumulators::binning_analysis_tag, alps::accumulators::impl::Accumulator<std::__1::vector<long double, std::__1::allocator<long double> >, alps::accumulators::error_tag, alps::accumulators::impl::Accumulator<std::__1::vector<long double, std::__1::allocator<long double> >, alps::accumulators::mean_tag, alps::accumulators::impl::Accumulator<std::__1::vector<long double, std::__1::allocator<long double> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<long double, std::__1::allocator<long double> > > > > > >::collective_merge(alps::mpi::communicator const&, int) const in binning_analysis.cpp.o
  "void alps::alps_mpi::reduce<unsigned long long, std::__1::plus<unsigned long long> >(alps::mpi::communicator const&, unsigned long long const&, unsigned long long&, std::__1::plus<unsigned long long>, int)", referenced from:
      alps::accumulators::impl::Accumulator<float, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<float> >::collective_merge(alps::mpi::communicator const&, int) in count.cpp.o
      alps::accumulators::impl::Accumulator<double, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<double> >::collective_merge(alps::mpi::communicator const&, int) in count.cpp.o
      alps::accumulators::impl::Accumulator<long double, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<long double> >::collective_merge(alps::mpi::communicator const&, int) in count.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<float, std::__1::allocator<float> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<float, std::__1::allocator<float> > > >::collective_merge(alps::mpi::communicator const&, int) in count.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<double, std::__1::allocator<double> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<double, std::__1::allocator<double> > > >::collective_merge(alps::mpi::communicator const&, int) in count.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<long double, std::__1::allocator<long double> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<long double, std::__1::allocator<long double> > > >::collective_merge(alps::mpi::communicator const&, int) in count.cpp.o
  "void alps::alps_mpi::reduce<unsigned long long, std::__1::plus<unsigned long long> >(alps::mpi::communicator const&, unsigned long long const&, std::__1::plus<unsigned long long>, int)", referenced from:
      alps::accumulators::impl::Accumulator<float, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<float> >::collective_merge(alps::mpi::communicator const&, int) const in count.cpp.o
      alps::accumulators::impl::Accumulator<double, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<double> >::collective_merge(alps::mpi::communicator const&, int) const in count.cpp.o
      alps::accumulators::impl::Accumulator<long double, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<long double> >::collective_merge(alps::mpi::communicator const&, int) const in count.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<float, std::__1::allocator<float> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<float, std::__1::allocator<float> > > >::collective_merge(alps::mpi::communicator const&, int) const in count.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<double, std::__1::allocator<double> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<double, std::__1::allocator<double> > > >::collective_merge(alps::mpi::communicator const&, int) const in count.cpp.o
      alps::accumulators::impl::Accumulator<std::__1::vector<long double, std::__1::allocator<long double> >, alps::accumulators::count_tag, alps::accumulators::impl::AccumulatorBase<std::__1::vector<long double, std::__1::allocator<long double> > > >::collective_merge(alps::mpi::communicator const&, int) const in count.cpp.o
ld: symbol(s) not found for architecture x86_64
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
```

File: accumulators/include/alps/accumulators/feature.hpp [hunk 1]
```diff
@@ -128,22 +130,22 @@ namespace alps {
...
-                        throw std::runtime_error("The Function operator *= is not implemented for accumulators, only for results" + ALPS_STACKTRACE); 
+                        throw std::runtime_error("The Function operator *= is not implemented for accumulators, only for results" + ALPS_STACKTRACE);
                     }
                     template<typename U> void operator/=(U) {
-                        throw std::runtime_error("The Function operator /= is not implemented for accumulators, only for results" + ALPS_STACKTRACE); 
+                        throw std::runtime_error("The Function operator /= is not implemented for accumulators, only for results" + ALPS_STACKTRACE);
                     }
                     void negate() {
-                        throw std::runtime_error("The Function gegate is not implemented for accumulators, only for results" + ALPS_STACKTRACE); 
+                        throw std::runtime_error("The Function gegate is not implemented for accumulators, only for results" + ALPS_STACKTRACE);
```
Comment (egull) [inline accumulators/include/alps/accumulators/feature.hpp]: could you please change gegate to negate?

File: accumulators/include/alps/accumulators/feature/count.hpp [hunk 2]
```diff
@@ -91,82 +87,64 @@ namespace alps {
...
-                    template<typename U> void augadd(U const & arg, typename boost::enable_if<boost::is_scalar<U>, int>::type = 0) {
+                    template<typename U> void augadd(U const & arg, typename std::enable_if<std::is_scalar<U>::value, int>::type = 0) {
                         if (m_count == 0)
                             throw std::runtime_error("The results needs measurements" + ALPS_STACKTRACE);
                         B::operator+=(arg);
                     }
-                    template<typename U> void augadd(U const & arg, typename boost::disable_if<boost::is_scalar<U>, int>::type = 0) {
+                    template<typename U> void augadd(U const & arg, typename std::enable_if<!std::is_scalar<U>::value, int>::type = 0) {
                         if (m_count == 0 || arg.count() == 0)
                             throw std::runtime_error("Both results needs measurements" + ALPS_STACKTRACE);
```
Comment (egull) [inline accumulators/include/alps/accumulators/feature/count.hpp]: need, not needs (and yes, I know it's not your mistake... just happened to find it)

File: accumulators/include/alps/accumulators/feature/count.hpp [hunk 3]
```diff
@@ -91,82 +87,64 @@ namespace alps {
...
-                    }
+                    void reset();
 
                 private:
 
                     // TODO: make macro ...
-                    template<typename U> void augadd(U const & arg, typename boost::enable_if<boost::is_scalar<U>, int>::type = 0) {
+                    template<typename U> void augadd(U const & arg, typename std::enable_if<std::is_scalar<U>::value, int>::type = 0) {
                         if (m_count == 0)
                             throw std::runtime_error("The results needs measurements" + ALPS_STACKTRACE);
```
Comment (egull) [inline accumulators/include/alps/accumulators/feature/count.hpp]: results need measurements

File: accumulators/include/alps/accumulators/feature/count.hpp [hunk 4]
```diff
@@ -91,82 +87,64 @@ namespace alps {
...
                         if (m_count == 0 || arg.count() == 0)
                             throw std::runtime_error("Both results needs measurements" + ALPS_STACKTRACE);
                         m_count = std::min(m_count,  arg.count());
                         B::operator+=(arg);
                     }
 
-                    template<typename U> void augsub(U const & arg, typename boost::enable_if<boost::is_scalar<U>, int>::type = 0) {
+                    template<typename U> void augsub(U const & arg, typename std::enable_if<std::is_scalar<U>::value, int>::type = 0) {
                         if (m_count == 0)
                             throw std::runtime_error("The results needs measurements" + ALPS_STACKTRACE);
```
Comment (egull) [inline accumulators/include/alps/accumulators/feature/count.hpp]: need

**Comment by krivenko** [comment]
@kleinhenz @egull This must be something mac-specific... 
Did you wipe out the build directory before trying this PR? If so, we will need to discuss this issue tomorrow.

**Comment by egull** [comment]
I did wipe – let's look at it tomorrow.

**Comment by krivenko** [comment]
@kleinhenz @egull Chances are, the linker error is gone with my latest commit.

**Comment by egull** [comment]
nope, I'm getting the same still.

**Comment by kleinhenz** [comment]
same for me

**Comment by kleinhenz** [comment]
changing `std::size_t` to `unsigned long long` fixes it for me.

File: accumulators/src/mpi.cpp [hunk 5]
```diff
@@ -265,7 +265,7 @@
                 template void reduce(const communicator &, std::vector<std::vector<T>> const&, std::plus<T>, int);                                \
                 template void reduce(const communicator &, std::vector<std::vector<T>> const&, std::vector<std::vector<T>>&, std::plus<T>, int);
 
-            ALPS_INST_MPI_REDUCE(unsigned long)
+            ALPS_INST_MPI_REDUCE(std::size_t)
```
Comment (kleinhenz) [inline accumulators/src/mpi.cpp]: with `clang` on `macos` this needs to be `unsigned long long` to avoid linker errors.

**Comment by krivenko** [comment]
@kleinhenz I'd rather first understand why different instantiations are needed on different platforms...

**Comment by krivenko** [comment]
@kleinhenz Thank you, Joe! Yes, this makes a lot of sense.

**Comment by kleinhenz** [comment]
Let's wait to merge until we figure out Alex's linker error

**Comment by galexv** [comment]
I am going to merge it with "devel" branch so that it will be tested by CI; meantime i will figure out what's wrong with clang compilation on my laptop (AFAICS, it is a separate issue from https://github.com/ALPSCore/ALPSCore/pull/464#issuecomment-366585608).

**Comment by krivenko** [comment]
Did anyone try to build ALPSCore applications on top of this PR?

**Comment by galexv** [comment]
@krivenko I've just built tutorials (successfully), but not a real application.
@egull  Shall i quickly set up a Jenkins job to build maxent from "devel" branch?

> Let's wait to merge until we figure out Alex's linker error

It's resolved, was my mistake.

**Comment by krivenko** [comment]
> Looks like Maxent builds successfully with the new code.

Thank you, Alex. Could you please do the same for CT-HYB?
I see recent Jenkins builds of CT-HYB have been failing due to what appears to be a memory exhaustion. Maybe this problem disappears with the less header-oriented accumulators and hdf5...

**Comment by galexv** [comment]
I did it with CT-HYB, but the compilation still fails. It looks like it hits ~1GB memory mark:

> `Building CXX object CMakeFiles/alpscore_cthyb.dir/src/solver_real.cpp.o`
> `/home/jenkins/openmpi-1.10.1/bin/mpic++   -DUSE_QUAD_PRECISION -Dalpscore_cthyb_EXPORTS -I/scratch/jenkins/workspace/cthyb-alpscore-devel/build -I/scratch/jenkins/workspace/cthyb-alpscore-devel -I/scratch/jenkins/workspace/cthyb-alpscore-devel/include -I/scratch/jenkins/workspace/cthyb-alpscore-devel/test -isystem /home/jenkins/boost_1_60_0_b1/include -isystem /home/jenkins/eigen-eigen-5a0156e40feb -isystem /scratch/jenkins/workspace/cthyb-alpscore-devel/ALPSCore/install/include -isystem /home/jenkins/hdf5-1.8.13/include  --param ggc-min-expand=20 --param ggc-min-heapsize=32768 -DALPS_GF_DEBUG -g -fPIC   -std=gnu++11 -o CMakeFiles/alpscore_cthyb.dir/src/solver_real.cpp.o -c /scratch/jenkins/workspace/cthyb-alpscore-devel/src/solver_real.cpp`
> `g++: internal compiler error: Killed (program cc1plus)`
[...]
> `18.26user 1.97system 0:20.29elapsed 99%CPU (0avgtext+0avgdata 900748maxresident)k
396324inputs+36896outputs (39654major+283315minor)pagefaults 0swaps`

**Comment by egull** [comment]
Yeah. Don't worry about the jenkins problem of the CT-HYB. We'll try to fix this with @shinaoka once he's here, or let it be fixed by having a machine with more memory. Not worth putting resources into it right now.

**Pull request merged**